flink-ml-lib/src/test/java/org/apache/flink/ml/recommendation/SwingTest.java

```diff
@@ -0,0 +1,278 @@
+                                Row.of(11L, "10,0.058845768947156235;12,0.058845768947156235"),
+                                Row.of(
+                                        12L,
+                                        "13,0.09134833828228624;10,0.058845768947156235;11,0.058845768947156235"),
+                                Row.of(13L, "12,0.09134833828228624")));
+
+        results.sort(Comparator.comparing(o -> o.getFieldAs(0)));
+
+        for (int i = 0; i < results.size(); i++) {
+            Row result = results.get(i);
+            String itemRankScore = result.getFieldAs(1);
+            Row expect = expectedScoreRows.get(i);
+            assertEquals(expect.getField(0), result.getField(0));
+            assertEquals(expect.getField(1), itemRankScore);
+        }
+    }
+
+    @Test
+    public void testParam() {
+        Swing swing = new Swing();
+
+        assertEquals("item", swing.getItemCol());
+        assertEquals("user", swing.getUserCol());
+        assertEquals(100, swing.getK());
+        assertEquals(1000, swing.getMaxUserNumPerItem());
+        assertEquals(10, swing.getMinUserBehavior());
+        assertEquals(1000, swing.getMaxUserBehavior());
+        assertEquals(15, swing.getAlpha1());
+        assertEquals(0, swing.getAlpha2());
+        assertEquals(0.3, swing.getBeta(), 1e-9);
+
+        swing.setItemCol("item_1")
+                .setUserCol("user_1")
+                .setK(20)
+                .setMaxUserNumPerItem(500)
+                .setMinUserBehavior(10)
+                .setMaxUserBehavior(50)
+                .setAlpha1(5)
+                .setAlpha2(1)
+                .setBeta(0.35);
+
+        assertEquals("item_1", swing.getItemCol());
+        assertEquals("user_1", swing.getUserCol());
+        assertEquals(20, swing.getK());
+        assertEquals(500, swing.getMaxUserNumPerItem());
+        assertEquals(10, swing.getMinUserBehavior());
+        assertEquals(50, swing.getMaxUserBehavior());
+        assertEquals(5, swing.getAlpha1());
+        assertEquals(1, swing.getAlpha2());
+        assertEquals(0.35, swing.getBeta(), 1e-9);
+    }
+
+    @Test
+    public void testInputWithIllegalDataType() {
+        List<Row> rows =
+                new ArrayList<>(Arrays.asList(Row.of(0, "10"), Row.of(1, "11"), Row.of(2, "")));
+
+        DataStream<Row> dataStream =
+                env.fromCollection(
+                        rows,
+                        new RowTypeInfo(
+                                new TypeInformation[] {
+                                    BasicTypeInfo.LONG_TYPE_INFO, BasicTypeInfo.STRING_TYPE_INFO
+                                },
+                                new String[] {"user_id", "item_id"}));
+        Table data = tEnv.fromDataStream(dataStream);
+
+        try {
+            Table[] swingResultTables =
+                    new Swing()
+                            .setItemCol("item_id")
```

Comment:
Let's use the default value when possible, e.g., `itemCol`, `userCol` and `outputCol`. Using the default value would simplify the test and also shows how users would use the presented algorithms.

Same for other tests.